File: Fundamentais/Javascript/8. Conversão de bases.js
// Algoritmo Fundamental Nº 8
// Descrição : Conversão de base
// Problema  : Converter um número inteiro decimal para a
//             sua representação binária correspondente.
// Aplicações: Interpretação de dados e instruções armazenados em computador.

var n = parseInt(prompt("Digite um número inteiro decimal: "))
var q = n;
var r = [];
var i = 0;

while (q > 0) {
    r[i] = q % 2;
    i++;
    q = Math.trunc(q / 2);
}

document.write("Número na base binária = ");
for (var j = i - 1; j >= 0; j--) {
    document.write(r[j]);
}

File: Fundamentais/Python/8. Conversão de bases.py
# Algoritmo Fundamental Nº 8
# Descrição : Conversão de base
# Problema  : Converter um número inteiro decimal para a
#             sua representação binária correspondente.
# Aplicações: Interpretação de dados e instruções armazenados em computador.

import math

n = int(input("Digite um número inteiro decimal: "))
q = n
r = []
i = 0
a = 0

while (q > 0):
    r.insert(i, q % 2)
    i += 1
    q = math.trunc(q / 2)

j = i - 1
print("Número na base binária = ",end="")
for a in range(j, -1, -1):
    print(r[a], end="")
